src/cata_tiles.cpp

```diff
@@ -179,66 +193,212 @@ void cata_tiles::get_tile_information(std::string config_path, std::string &json
     }
 }
 
+inline pixel cata_tiles::get_pixel_color(SDL_Surface *surf, int x, int y, int w)
+{
+    pixel pix;
+    const auto pixelarray = reinterpret_cast<unsigned char *>(surf->pixels);
+    const auto pixel_ptr = pixelarray + (y * w + x) * 4;
+    pix.r = pixel_ptr[0];
+    pix.g = pixel_ptr[1];
+    pix.b = pixel_ptr[2];
+    pix.a = pixel_ptr[3];
+    return pix;
+}
+
+inline void cata_tiles::set_pixel_color(SDL_Surface *surf, int x, int y, int w, pixel pix)
+{
+    const auto pixelarray = reinterpret_cast<unsigned char *>(surf->pixels);
+    const auto pixel_ptr = pixelarray + (y * w + x) * 4;
+    pixel_ptr[0] = static_cast<unsigned char>(pix.r);
+    pixel_ptr[1] = static_cast<unsigned char>(pix.g);
+    pixel_ptr[2] = static_cast<unsigned char>(pix.b);
+    pixel_ptr[3] = static_cast<unsigned char>(pix.a);
+}
+
+void cata_tiles::convert_surface_to_grayscale(SDL_Surface *surf)
```

Comment:
These three methods seem like good candidates to be deduplicated, you can extract the code for iterating over the pixel values, and just have the logic that is applied to each pixel be declared seperately.